### DISCUSSION TITLE
UI Revamp: help us navigate font licenses to pick the right font for CKAN

### DISCUSSION THREAD

**Discussion by thegostev** [ckan/ckan#8486] [CKAN 3.0]
Hi,

The issue that we faced when implementing Noto Sans that was selected during the design stage is that, if using this font for a software, a data should be sent to Google continuously, which is unacceptable for CKAN as it's a non-opensource-friendly move and has GDPR concerns.

**Please suggest suitable fonts or font families that'd work for an open source software product.**

Requirements:

- The font should have an open-source license compatible with CKAN's licensing terms
- The font should be highly readable at various sizes, especially for long-form content.
- It should include a wide range of characters, including Latin, Cyrillic, Arabic, CJK.
- Similar characters (like 1, l, and I) should be easily distinguishable.
- Well-designed punctuation marks for improved readability in dense text.
- The font should be easy on the eyes for extended reading sessions.

I'd be happy to collaborate and discuss different option you'd suggest.

![rs=w_1280](https://github.com/user-attachments/assets/9a2bcdb7-cdea-49f4-b1a2-440028399115)

### More details on the issue with Noto Sans

Noto Sans is licensed under the SIL Open Font License (OFL), version 1.1. Which has a bundling requirement: 

> If you distribute Noto Sans with your software, you must include the entire font, including all its scripts and variations. You can't subset the font without changing its name.

When a website uses Google Fonts via the Google Fonts API (their CDN), some user data is indeed sent to Google's servers. This typically includes:

- The user's IP address
- The URL of the page where the font is being loaded
- Details about the user's browser and operating system

As the result [in 2022, a German court ruled](https://dxpr.com/drupal-blog/google-fonts-privacy-update-dxpr-theme#:~:text=In%20fact%2C%20German%20courts%20have,loading%20Google%20fonts%20this%20way.) that using Google Fonts via the CDN without user consent could violate GDPR (General Data Protection Regulation) due to the transfer of IP addresses to Google's servers.

**As an alternative we can self-host the font files on own servers which is not optimal, but is up for discussion.**

**Comment by wardi** [comment]
What's the problem with including a copy of the whole font with all its scripts and variations as static content as part of ckan?
